value/mapunstructured.go

```diff
@@ -72,14 +80,12 @@ func (m mapUnstructuredInterface) Equals(other Map) bool {
 		if !ok {
 			return false
 		}
-		vv := NewValueInterface(v)
+		vv.Value = v
```

Comment:
oops, forgot to optimize this one like I had for the other map implementations. Done.

We might want to do the `GetInto/AtInto` optimization as a future improvement It would help in a few ways, like eliminating the need for having custom Equals functions on each map implementation. I'd like to pursue it in a future PR since it's not entirely trivial Callers that are using the map interface don't know if the Into functions are using a underlying `valueReflect` or a `valueUnstructured` so the signature would need to be something like: `GetInto(key string, ValueReuser)` or something where `ValueReuser` (we probably wouldn't call it that) would be able to reuse either a `valueReflect` or a `valueUnstructured` as needed.